### PR TITLE
Faster position matrix

### PR DESCRIPTION
**Pull request by gitpeterwind**
Faster position matrix, specially for large number of MPI. 
Orbitals are sent 6 times less, by reusing orbitals for x y and z and symmetry. Will give better cache utilization also.
NB: this commit will not change results compared to previous commit, but the results are *not* the same as for a couple of weeks ago (HF energy differs by 10%).

### CODE REVIEW

File: src/utils/RRMaximizer.cpp [hunk 1]
```diff
@@ -55,9 +56,47 @@ RRMaximizer::RRMaximizer(double prec, OrbitalVector &Phi) {
...
+                    R_z(idx_i, idx_j) = orbital::dot(bra_i, ket_j_z);
+                    R_z(idx_j, idx_i) = R_z(idx_i, idx_j);
+                }
+            }
+        }
+    }
+
+    //    mpi::allreduce_matrix(R_x, mpi::comm_orb);
+    //    mpi::allreduce_matrix(R_y, mpi::comm_orb);
+    //    mpi::allreduce_matrix(R_z, mpi::comm_orb);
```
Comment (stigrj) [inline src/utils/RRMaximizer.cpp]: Can you remove these lines?

File: src/utils/RRMaximizer.cpp [hunk 2]
```diff
@@ -55,9 +56,47 @@ RRMaximizer::RRMaximizer(double prec, OrbitalVector &Phi) {
...
-    ComplexMatrix R_x = r_x(Phi, Phi);
-    ComplexMatrix R_y = r_y(Phi, Phi);
-    ComplexMatrix R_z = r_z(Phi, Phi);
+    ComplexMatrix R_x = ComplexMatrix::Zero(Phi.size(),Phi.size());
+    ComplexMatrix R_y = ComplexMatrix::Zero(Phi.size(),Phi.size());
+    ComplexMatrix R_z = ComplexMatrix::Zero(Phi.size(),Phi.size());
+
+    OrbitalVector Phi_r_x_Vec = r_x(Phi);
+    OrbitalVector Phi_r_y_Vec = r_y(Phi);
+    OrbitalVector Phi_r_z_Vec = r_z(Phi);
```
Comment (stigrj) [inline src/utils/RRMaximizer.cpp]: Name suggestions for these vectors :slightly_smiling_face: :
```
OrbitalVector xPhi = r_x(Phi);
OrbitalVector yPhi = r_y(Phi);
OrbitalVector zPhi = r_z(Phi);
```

Comment (gitpeterwind) [reply src/utils/RRMaximizer.cpp]: Yes, nicer that way!

**Comment by stigrj** [comment]
The CircleCI `mpi-py3` run sometimes hangs in the last integration test (Li polarizability)

**Pull request merged**